Moving adapter behavior changes (#6151)
fishtown-analytics/docs.getdbt.com: website/docs/reference/global-configs/redshift-changes.md

---

> title: "Redshift adapter behavior changes"

Reviewer: ```suggestion
title: "Amazon Redshift adapter behavior changes"
```

id: "redshift-changes"
sidebar: "Redshift"
---

> ## restrict_direct_pg_catalog_access

Reviewer: ```suggestion
## The restrict_direct_pg_catalog_access flag
```

> Originally, the `dbt-redshift` adapter was built on top of the `dbt-postgres` adapter and used Postgres tables for metadata access. With this flag enabled, the adapter will use the Redshift API (through the Python client) if available, or query Redshift's `information_schema` tables instead of using `pg_` tables.

Reviewer: ```suggestion
Originally, the `dbt-redshift` adapter was built on top of the `dbt-postgres` adapter and used Postgres tables for metadata access. When this flag is enabled, the adapter uses the Redshift API (through the Python client) if available, or queries Redshift's `information_schema` tables instead of using the `pg_` tables.
```

> While we don't expect any user-noticeable behavior changes due to this change, out of caution we are gating it behind a behavior-change flag and encouraging users to test it before it becomes the default for everyone.

Reviewer: ```suggestion
While you shouldn't notice any behavior changes due to this change, however, to be cautious dbt Labs is gating it behind a behavior-change flag and encouraging you to test it before it becoming the default.
```
